src/sentry/models/projectownership.py

```diff
@@ -53,18 +53,52 @@ def get_owners(cls, project_id, data):
                 project_id=project_id,
             )
 
+        rules = cls._matching_ownership_rules(ownership, project_id, data)
+        if not rules:
+            return cls.Everyone if ownership.fallthrough else [], None
+
+        owners = {o for rule in rules for o in rule.owners}
+        return filter(None, resolve_actors(owners, project_id).values()), rules
+
+    @classmethod
+    def get_autoassign_owner(cls, project_id, data):
+        """
+        Get the auto-assign owner for a project if there are any.
+
+        Will return None if there are no owners, or a list of owners.
+        """
+        try:
+            ownership = cls.objects.get(project_id=project_id)
+        except cls.DoesNotExist:
+            return None
+        if not ownership.auto_assignment:
+            return None
+
+        rules = cls._matching_ownership_rules(ownership, project_id, data)
+        if not rules:
+            return None
+
+        score = 0
+        owners = None
+        # Automatic assignment prefers the owner with the longest
+        # matching pattern as the match is more specific.
+        for rule in rules:
+            candidate = len(rule.matcher.pattern)
+            if candidate > score:
+                score = candidate
+                owners = rule.owners
+        actors = filter(None, resolve_actors(owners, project_id).values())
+        return actors[0].resolve()
```

Comment:
Any chance the list will be empty here?

Reply:
I don't think so. Each rule needs to have at least one owner to be a valid rule.